Tools/Scripts/webkitpy/layout_tests/controllers/layout_test_finder_unittest.py

```diff
@@ -0,0 +1,131 @@
+        self.assertEqual([("", "a", "")], v)
+
+        v = list(self.finder._split_glob("a/"))
+        self.assertEqual([("a", "", "")], v)
+
+        v = list(self.finder._split_glob("a/b"))
+        self.assertEqual([("a", "b", "")], v)
+
+        v = list(self.finder._split_glob("a/b/"))
+        self.assertEqual([("a/b", "", "")], v)
+
+        v = list(self.finder._split_glob("a/b/c"))
+        self.assertEqual([("a/b", "c", "")], v)
+
+        v = list(self.finder._split_glob("a/b#c"))
+        self.assertEqual([("a", "b", "#c")], v)
+
+        v = list(self.finder._split_glob("a/b?c"))
+        self.assertEqual([("a", "b", "?c"), ("a", "b?c", "")], v)
+
+        v = list(self.finder._split_glob("a/b?c?d"))
+        self.assertEqual(
+            [("a", "b", "?c?d"), ("a", "b?c", "?d"), ("a", "b?c?d", "")], v
+        )
+
+        v = list(self.finder._split_glob("a/b?c#d"))
+        self.assertEqual([("a", "b", "?c#d"), ("a", "b?c", "#d")], v)
+
+        v = list(self.finder._split_glob("a/?b"))
+        self.assertEqual([("a", "?b", "")], v)
+
+        v = list(self.finder._split_glob("a/?b?c"))
+        self.assertEqual([("a", "?b", "?c"), ("a", "?b?c", "")], v)
+
+        v = list(self.finder._split_glob("a?b"))
+        self.assertEqual([("", "a", "?b"), ("", "a?b", "")], v)
+
+        v = list(self.finder._split_glob("a??b"))
+        self.assertEqual([("", "a", "??b"), ("", "a?", "?b"), ("", "a??b", "")], v)
+
+        v = list(self.finder._split_glob("a?b?c"))
+        self.assertEqual([("", "a", "?b?c"), ("", "a?b", "?c"), ("", "a?b?c", "")], v)
+
+        v = list(self.finder._split_glob("a/#b"))
+        self.assertEqual([], v)
+
+    def test_is_test_file(self):
+        finder = self.finder
+        self.assertTrue(finder.is_test_file("", "foo.html"))
+        self.assertTrue(finder.is_test_file("", "foo.shtml"))
+        self.assertTrue(finder.is_test_file("", "foo.svg"))
+        self.assertTrue(finder.is_test_file("", "test-ref-test.html"))
+        self.assertFalse(finder.is_test_file("", "foo.png"))
+        self.assertFalse(finder.is_test_file("", "foo-expected.html"))
+        self.assertFalse(finder.is_test_file("", "foo-expected.svg"))
+        self.assertFalse(finder.is_test_file("", "foo-expected.xht"))
+        self.assertFalse(finder.is_test_file("", "foo-expected-mismatch.html"))
+        self.assertFalse(finder.is_test_file("", "foo-expected-mismatch.svg"))
+        self.assertFalse(finder.is_test_file("", "foo-expected-mismatch.xhtml"))
+        self.assertFalse(finder.is_test_file("", "foo-ref.html"))
+        self.assertFalse(finder.is_test_file("", "foo-notref.html"))
+        self.assertFalse(finder.is_test_file("", "foo-notref.xht"))
+        self.assertFalse(finder.is_test_file("", "foo-ref.xhtml"))
+        self.assertFalse(finder.is_test_file("", "ref-foo.html"))
+        self.assertFalse(finder.is_test_file("", "notref-foo.xhr"))
+        self.assertFalse(finder.is_test_file("", "foo_wsh.py"))
```

Comment:
We have some Python CGI scripts, right? That means `is_test_file` isn't quite doing what I'd expect.

Reply:
https://github.com/WebKit/WebKit/blob/3412f17868274cc523f247c8dbc336a73d273abe/Tools/Scripts/webkitpy/layout_tests/controllers/layout_test_finder_legacy.py#L48 — this is a special-case

Reply:
Should we have a `.py` file in here to make it clear that `.py` files _are_ a valid test name?